samples/RawPubSub/src/main/java/rawpubsub/RawPubSub.java

```diff
@@ -0,0 +1,252 @@
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package rawpubsub;
+
+import software.amazon.awssdk.crt.CRT;
+import software.amazon.awssdk.crt.CrtRuntimeException;
+import software.amazon.awssdk.crt.io.ClientBootstrap;
+import software.amazon.awssdk.crt.io.TlsContext;
+import software.amazon.awssdk.crt.io.TlsContextOptions;
+import software.amazon.awssdk.crt.mqtt.MqttClient;
+import software.amazon.awssdk.crt.mqtt.MqttClientConnection;
+import software.amazon.awssdk.crt.mqtt.MqttClientConnectionEvents;
+import software.amazon.awssdk.crt.mqtt.MqttMessage;
+import software.amazon.awssdk.crt.mqtt.QualityOfService;
+import software.amazon.awssdk.iot.iotjobs.model.RejectedError;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+class RawPubSub {
+    static String clientId = "samples-client-id";
+    static String rootCaPath;
+    static String certPath;
+    static String keyPath;
+    static String endpoint;
+    static String topic = "/samples/test";
+    static String message = "Hello World!";
+    static int    messagesToPublish = 10;
+    static boolean showHelp = false;
+
+    static String userName;
+    static String password;
+    static String protocolName = "mqtt";
+    static List<String> authParams;
+
+    static void printUsage() {
+        System.out.println(
+                "Usage:\n"+
+                "  --help        This message\n"+
+                "  --clientId    Client ID to use when connecting (optional)\n"+
+                "  -e|--endpoint AWS IoT service endpoint hostname\n"+
+                "  -r|--rootca   Path to the root certificate\n"+
+                "  -c|--cert     Path to the IoT thing certificate\n"+
+                "  -k|--key      Path to the IoT thing public key\n"+
+                "  -t|--topic    Topic to subscribe/publish to (optional)\n"+
+                "  -m|--message  Message to publish (optional)\n"+
+                "  -n|--count    Number of messages to publish (optional)"+
+                "  -u|--username Username to use as part of the connection/authentication process\n"+
+                "  --password    Password to use as part of the connection/authentication process\n"+
+                "  --protocol    (optional) Communication protocol to use; defaults to mqtt\n"+
+                "  --auth_params (optional) Comma delimited list of auth parameters. For websockets these will be set as headers.  For raw mqtt these will be appended to user_name.\n"
+        );
+    }
+
+    static void parseCommandLine(String[] args) {
```

Comment:
Can we at some point move most of this to some central thing somewhere that returns a config object, so we don't have to copy-pasta to every sample?

Reply:
That'd sure be nice